### ISSUE TITLE
Mitsubishi PEAD71

### ISSUE THREAD

**Issue by PhineousFreak** [scottyphillips/echonetlite_homeassistant#76]
Apologies in advance if this request is not the correct use of the "issues" facility .. this is all new to me

I have a ducted PEAD system & from the wired Mitsubishi controller the temperature range in heat is from 17C upwards.  But from the iphone App the range is from 10C upwards.

My system install is odd & I need to use the return duct temperature rather than room temperature, plus the return point is well away from the head outlets & therefore the usual setpoint I use is 13C. 

I managed to get the echolite add in operating to 10C by changing line 243 of _init_.py to             'min_temp_heat': 10,

But the issue I have is that the thermostat card in lovelace still only gives a range from 16C on the front dial.  I can override this on the details tab, but it would be nice to have the full range available from the front dial.  

My coding skills are meagre & I have not figured out what else I would need to edit to solve this issue.  I suspect it may be something to do with the temperature range that the heatpump sends to the addin when interrogated, but that is just a wild guess!

Any help to resolve approciated

Otherwise, thanks for the integration it is excellent.  

As a feature request, would there be anyway to override the heatpumps return/room temperature sensors from an external device via the add-in?

**Comment by scottyphillips** [comment]
Back under integrations, under echonetlite there is a ‘configure’ button where you can tweak stuff. I created a way of configuring minimum and maximum temperatures using sliders under different scenarios like heating cooling auto. I probably just need to tweak the setting for minimum heat to permit 10 degree operation? Would that work for you?

![B4013346-9111-4BBB-BC89-63C97423C0AD](https://user-images.githubusercontent.com/26150540/173479913-8ddc9ff6-739e-418e-bb93-fbfc380406bc.png)

**Comment by PhineousFreak** [comment]
Hi Scotty

I am not sure.

If I look at the climate.heatpump entity that was created off my install of the integration the yaml is:-

hvac_modes:
  - heat
  - cool
  - dry
  - fan_only
  - heat_cool
  - 'off'
min_temp: 15
max_temp: 18
target_temp_step: 1
fan_modes:
  - low
  - medium-low
  - medium-high
current_temperature: 13
temperature: 16
fan_mode: medium-low
hvac_action: 'off'
friendly_name: Heatpump
supported_features: 9

I can change the min_temp and max_temp to (say) 10 and 28 & the thermostat card briefly works over that range.  But then reverts to the range 15-18 after a couple of seconds.

In the config file for Echonet the sliders are there and have a range 10C - 30C for heating already.  As I indicated in my initial post I had edited the init.py to 'min_temp_heat': 10 so maybe that is why the range is already correct - by maybe I also needed to tweak something else apart from that line ?

**Comment by scottyphillips** [comment]
Go to ‘const.py’ and update min_temp_heat value from 15 to 10. 
```
TEMP_OPTIONS = {"min_temp_heat": {"min":15, "max":25},
                "max_temp_heat": {"min":18, "max":30},
                "min_temp_cool": {"min":15, "max":25},
                "max_temp_cool": {"min":18, "max":30},
                "min_temp_auto": {"min":15, "max":25},
                "max_temp_auto": {"min":18, "max":30},
}
```

**Comment by scottyphillips** [comment]
![Uploading ACF189B9-C24E-4FD9-9797-59C2D416B59D.jpeg…]()

**Comment by scottyphillips** [comment]
![12696837-87E6-4E2A-8C4B-98AE1BE9F974](https://user-images.githubusercontent.com/26150540/173494623-29e479bd-d395-4c63-a579-5b81f5698a0b.png)

**Comment by scottyphillips** [comment]
![Uploading 4F911FF3-5591-427A-9A44-C0F4DA1CCFBC.jpeg…]()

**Comment by scottyphillips** [comment]
![1C9340F2-62EF-4E3B-81FB-167F3AC19032](https://user-images.githubusercontent.com/26150540/173494780-8a820f17-2ad3-4584-9747-95629092ca44.jpeg)

**Comment by scottyphillips** [comment]
![2D7D720F-5351-4142-BE39-EE513E751677](https://user-images.githubusercontent.com/26150540/173494816-5302783c-57d7-42aa-999d-e2eda66209ce.png)

**Comment by PhineousFreak** [comment]
Looks like I had already changed the const.py

TEMP_OPTIONS = {"min_temp_heat": {"min":10, "max":25},

**Comment by PhineousFreak** [comment]
I had wondered if it was the climate.py 

specifically

    _@property
    def min_temp(self) -> int:
        """Return the minimum temperature supported by the HVAC."""
        if self.hvac_mode == HVAC_MODE_HEAT:
            self._min_temp = self._connector._user_options['min_temp_heat']
        if self.hvac_mode == HVAC_MODE_COOL:
            self._min_temp = self._connector._user_options['min_temp_cool']
        if self.hvac_mode == HVAC_MODE_HEAT_COOL:
            self._min_temp = self._connector._user_options['min_temp_auto']
        return self._min_temp_


And whether that overwrites the other changes?  But of course I don't really know what I am looking at/doing :-)

**Comment by scottyphillips** [comment]
I am not seeing any issues.
I updated the min_temp_heat value from 15 to 10 in both const.py and init.py. I rebooted home assistant. I then went to the slider and adjusted the value to 10 and saved the updated value

![78863B57-52F0-4F82-A454-A2AE3B5DD49C](https://user-images.githubusercontent.com/26150540/173499554-ea66a1d7-4354-4af3-9a1d-25a11800fe95.jpeg)

And then after a short while the graphic updated. It doesn’t change back.
![1013DBB8-28D3-4509-BF79-3B41D52E75A6](https://user-images.githubusercontent.com/26150540/173499637-288b663a-41c3-41f5-bc11-352d08e5abc6.jpeg)
